client/src/routes/AdminRoutes.js

```diff
@@ -25,7 +26,8 @@ const childRoutes = [
 
     { path: '/edit-user/:id'     , component: UserFormPage },
     { path: '/edit-category/:id' , component: CategoryFormPage },
-    { path: '/edit-service/:id'  , component: ServiceFormPage }
+    { path: '/edit-service/:id'  , component: ServiceFormPage },
+    { path: '/visualize-data', component: DataVisualizationPage}
```

Comment:
There is a pattern of spacing for the route files that should be obvious, please follow it. Let me know if you don't understand what I mean.